src/Storage/Commands.Storage/Common/Cmdlet/NewAzureStorageContext.cs

```diff
@@ -281,6 +316,77 @@ internal CloudStorageAccount GetAnonymousStorageAccount(string storageAccountNam
             return GetStorageAccountWithAzureEnvironment(credential, storageAccountName, useHttps, azureEnvironmentName);
         }
 
+        /// <summary>
+        /// Get OAuth storage account
+        /// </summary>
+        /// <param name="storageAccountName">Storage account name, it's used for build end point</param>
+        /// <param name="useHttps"></param>
+        /// <param name="storageEndpoint"></param>
+        /// <returns>A storage account</returns>
+        internal CloudStorageAccount GetStorageAccountByOAuth(string storageAccountName, bool useHttps, string storageEndpoint = "")
+        {
+            var tokenStr = GetOAuthToken();
+            TokenCredential tokenCredential = new TokenCredential(tokenStr, TokenRenewer, null, new TimeSpan(0, 1, 0));
+            StorageCredentials credential = new StorageCredentials(tokenCredential);
+            return GetStorageAccountWithEndPoint(credential, storageAccountName, useHttps, storageEndpoint);
+        }
+
+        /// <summary>
+        /// Get OAuth storage account
+        /// </summary>
+        /// <param name="storageAccountName">Storage account name, it's used for build end point</param>
+        /// <param name="useHttps"></param>
+        /// <param name="storageEndpoint"></param>
+        /// <returns>A storage account</returns>
+        internal CloudStorageAccount GetStorageAccountByOAuthFromAzureEnvironment(string storageAccountName, bool useHttps, string azureEnvironmentName = "")
+        {
+            var tokenStr = GetOAuthToken();
+            TokenCredential tokenCredential = new TokenCredential(tokenStr, TokenRenewer, null, new TimeSpan(0, 1, 0));
+            StorageCredentials credential = new StorageCredentials(tokenCredential);
+            return GetStorageAccountWithAzureEnvironment(credential, storageAccountName, useHttps, azureEnvironmentName);
+        }
+        
+
+        /// <summary>
+        /// The Task to renew Oauth Token, it must be Async task since the new TokenCredential() need Async Task
+        /// </summary>
+        /// <returns>Microsoft.WindowsAzure.Storage.Auth.NewTokenAndDateTimeOffset which represents a new token and a new time offset.</returns>
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        private async Task<NewTokenAndFrequency> TokenRenewer(Object state, CancellationToken cancellationToken)
```

Comment:
I don't think this method needs to be async.  Also, the 'accessToken' variable in this should not come from the cmdlet - it should probably be stored in the context, or be a closure (for example, you could have a method on this cmdlet that returned a renewer).

Reply:
The XSCL define the renew function is async as below, so we must use async.

`public TokenCredential(string initialToken, RenewTokenFuncAsync periodicTokenRenewer, object state, TimeSpan renewFrequency);`

I will change the access token as closure.